### DISCUSSION TITLE
pubsub module

### DISCUSSION THREAD

**Discussion by noeldiviney** [msys2/MINGW-packages#16512] [General]
any plans to have a pubsub module

**Comment by Biswa96** [comment]
Would you like to explain what pubsub module is?

**Reply by noeldiviney** [reply]
Hi,

The following link is a pretty good explanation of python pubsub
https://www.blog.pythonlibrary.org/2019/03/28/wxpython-4-and-pubsub/
	
In reading this myself I found that pubsub is integrated into wxPython and removes the need to having pip install the pubsub module.
However I notice that pubsub is being deprecated and replaced by pypubsub.
I am now able to build my wxPython GUI and create a --onefile executeable.

I wish to express my appreciation to the MSYS2 team for the tremendous work in making this possible.
Thanks guys!!

Noel.

**Comment by noeldiviney** [comment]
Hi,

The following link is a pretty good explanation of python pubsub
https://www.blog.pythonlibrary.org/2019/03/28/wxpython-4-and-pubsub/
[https://www.blog.pythonlibrary.org/wp-content/uploads/2016/03/cropped-cropped-cropped-mvp-1.png]<https://www.blog.pythonlibrary.org/2019/03/28/wxpython-4-and-pubsub/>
wxPython 4 and PubSub - Mouse Vs Python<https://www.blog.pythonlibrary.org/2019/03/28/wxpython-4-and-pubsub/>
The Publish-Subscribe pattern is pretty common in computer science and very useful too. The wxPython GUI toolkit has had an implementation of it for a
www.blog.pythonlibrary.org
In reading this myself I found that pubsub is integrated into wxPython and removes the need to having pip install the pubsub module.
However I notice that pubsub is being deprecated and replaced by pypubsub.
I am now able to build my wxPython GUI and create a --onefile executeable.

I wish to express my appreciation th the MSYS2 team for the tremendous work in making this possible.
Thanks guys!!

Noel.


Noel Diviney Ayrenet Pty Ltd, Unit 3,80-88 Rodeo Drive, Dandenong South, Victoria 3175, Australia. Phone +61 413 120 892 email ***@***.*** or ***@***.***

________________________________

**Comment by Biswa96** [comment]
It seems that the pypubsub repository is not maintained anymore. The last commit in the https://github.com/schollii/pypubsub repository was in 1st April 2019. Old repository can not be added. But you can try to use the module from pypi or from git repo as usual.

**Reply by noeldiviney** [reply]
Thanks for the update, appreciated.
I have a question with regard to python modules and mys64 mingw64.
Is it ok to mix using pacman package manager and pip in the loading of modules?

**Reply by Biswa96** [reply]
There are some quirks. See https://www.msys2.org/docs/python/

**Reply by noeldiviney** [reply]
Many thanks.
I will use pacman unless there is no package available
Noel

**Reply by noeldiviney** [reply]
J have been doing some more testing.
I have installed msys2 to c::/msys64 as per the instructions and then run pacman -Syuu 3 times
I then installed python as "pacman -S mingw-w64-x86_64-python"
The attached  msys64Python.txt shows a log of my attempt to create a new folder
Everything seems to work but the folder is not created  
[msys64Python.txt](https://github.com/msys2/MINGW-packages/files/11128782/msys64Python.txt)

I have tested on linux Mint and the same tests work as expected see attached
[linux-python3-test.txt](https://github.com/msys2/MINGW-packages/files/11128796/linux-python3-test.txt)

Suggestions are appreciated.
Noel

**Reply by noeldiviney** [reply]
I have found the problem.
the Path "/home/eicon/jtagit/dload1" is not recognized to be the same as "jtagit/dload1".
Now the python command "os.path.exists("jtagit/dload1") works properly
